### ISSUE TITLE
Duplicate forum post submissions

### ISSUE THREAD

**Issue by HenrikLam** [Submitty/Submitty#6613]
**What problem are you trying to solve with Submitty**
While posting on a course forum, if you click any buttons that will submit a reply or a post multiple times, then multiple AJAX requests are sent to the server creating multiple of the same post/reply.

**Describe the way you'd like to solve this problem**
It was suggested by @williamjallen during meeting that we can disable the buttons once a submission button is clicked, and then reenabling it once the AJAX request is finished.

**Additional context**
Steps to recreate the bug

1. Go to the sample course on submitty and go to the discussion forum
2. Click create thread button
3. Enter arbitrary post title and post content and pick any categories
4. Click publish thread multiple times before the page is refreshed
5. See that duplicates of the same post is created on the discussion forum

The same issue is also seen similarly when replying to a thread.
